Scripts/build/Jenkins/Jenkinsfile

```diff
@@ -0,0 +1,665 @@
+    }
+    dir("${workspace}/${ENGINE_REPOSITORY_NAME}") {
+        // Get python
+        if(env.IS_UNIX) {
+            sh label: 'Getting python',
+               script: 'python/get_python.sh'
+        } else {
+            bat label: 'Getting python',
+                script: 'python/get_python.bat'
+        }
+
+        // Always run the clean step, the scripts detect what variables were set, but it also cleans if
+        // the NODE_LABEL has changed
+        def command = "${pipelineConfig.PYTHON_DIR}/python"
+        if(env.IS_UNIX) command += '.sh'
+        else command += '.cmd'
+        command += " -u ${pipelineConfig.BUILD_ENTRY_POINT} --platform ${platform} --type clean"
+        palSh(command, "Running ${platform} clean")
+    }
+}
+
+def Build(Map pipelineConfig, String platform, String type, String workspace) {
+    // If EXECUTE_FROM_PROJECT is defined, we execute the script from the project instead of from the engine
+    // In both cases, the scripts are in the engine, is just what the current dir is and how we get to the scripts
+    def currentDir = "${workspace}/${ENGINE_REPOSITORY_NAME}"
+    def pathToEngine = ""
+
+    timeout(time: env.TIMEOUT, unit: 'MINUTES', activity: true) {
+        def command = "${pipelineConfig.PYTHON_DIR}/python"
+        def ext = ''
+        if(env.IS_UNIX) {
+            command += '.sh'
+            ext = '.sh'
+        }
+        else command += '.cmd'
+
+        // Setup environment for project execution, otherwise, register the project
+        if (env.EXECUTE_FROM_PROJECT?.toBoolean()) {
+            currentDir = "${workspace}/${PROJECT_REPOSITORY_NAME}"
+            pathToEngine = "../${ENGINE_REPOSITORY_NAME}/"
+        } else {
+            dir("${workspace}/${ENGINE_REPOSITORY_NAME}") {
+                palSh("scripts/o3de${ext} register --project-path ${workspace}/${PROJECT_REPOSITORY_NAME}", "Registering project ${PROJECT_REPOSITORY_NAME}")
+            }
+        }
+        command += " -u ${pipelineConfig.BUILD_ENTRY_POINT} --platform ${platform} --type ${type}"
+        dir("${workspace}/${ENGINE_REPOSITORY_NAME}") {
+            palSh(command, "Running ${platform} ${type}")
+        }
+    }
+}
+
+def TestMetrics(Map pipelineConfig, String workspace, String branchName, String repoName, String buildJobName, String outputDirectory, String configuration) {
+    catchError(buildResult: null, stageResult: null) {
+        def cmakeBuildDir = [workspace, ENGINE_REPOSITORY_NAME, outputDirectory].join('/')
+        dir("${workspace}/${ENGINE_REPOSITORY_NAME}") {
+            checkout scm: [
+                $class: 'GitSCM',
+                branches: [[name: '*/main']],
+                extensions: [
+                    [$class: 'AuthorInChangelog'],
+                    [$class: 'RelativeTargetDirectory', relativeTargetDir: 'mars']
+                ],
+                userRemoteConfigs: [[url: "${env.MARS_REPO}", name: 'mars', credentialsId: "${env.GITHUB_USER}"]]

```

Comment:
nit: MARS should be optional. Do we need to onboard with MARS for this part to work?